include/swift/AST/DiagnosticsSema.def

```diff
@@ -663,6 +663,9 @@ ERROR(super_initializer_not_in_initializer,none,
       "'super.init' cannot be called outside of an initializer", ())
 WARNING(isa_is_always_true,none, "'%0' test is always true",
         (StringRef))
+WARNING(isa_is_foreign_check,none,
+      "'%0' test always succeeds because %1 is a foreign type",
```

Comment:
What do we do for `as!`? Do we just use the generic message?

Unless @rjmccall has any objections, I'd be okay with just hardcoding "Core Foundation type" for now. I think I'd also change the message to say *why* the check always returns true (and avoid "succeeds").